src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,181 @@
-import React from 'react';
+import React, { useState } from 'react';
+import {
+  addComments,
+  deleteComments,
+  getComments,
+  getPostComments,
+} from '../../api/posts';
+import { Comment } from '../../types/Comment';
+import { Post } from '../../types/Post';
+import { validation } from '../../utils/validation';
+import { Loader } from '../Loader';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectPostDetails: Post | null;
+  selectedPostId: number | null;
+};
+
+export const PostDetails: React.FC<Props> = (
+  {
+    selectPostDetails,
+    selectedPostId,
+  },
+) => {
+  const [postName, setPostName] = useState<string>('');
+  const [postEmail, setPostEmail] = useState<string>('');
+  const [postBody, setPostBody] = useState<string>('');
```

Comment:
you may create common state-object for post fields, create initial constant object with empty fields and use it when you reset fields

Reply:
=)